docs/rfcs/rfc-007-subscriptions.md

```diff
@@ -276,6 +276,135 @@ UNWIND meta AS m
+  moviesConnection(first: Int, after: String, where: MovieWhere, sort: [MovieSort]): MoviesConnection!
+}
+
+type Mutation {
+  createMovies(input: [MovieCreateInput!]!): CreateMoviesMutationResponse!
+  deleteMovies(where: MovieWhere): DeleteInfo!
+  updateMovies(where: MovieWhere, update: MovieUpdateInput): UpdateMoviesMutationResponse!
+}
+
+type Subscription {
+  movieCreated(where: MovieSubscriptionWhere): MovieCreatedEvent!
+  movieUpdated(where: MovieSubscriptionWhere): MovieUpdatedEvent!
+  movieDeleted(where: MovieSubscriptionWhere): MovieDeletedEvent!
+}
+```
+
+\
+\
+By excluding the `CREATE` and `DELETE` operations on the movie type like so:
+
+```graphql
+type Movie @exclude(operations: [CREATE, DELETE]) {
+  title: String!
+  released: Int!
+}
+```
+
+the auto-generated types will change to:
+
+```graphql
+type Query {
+  movies(where: MovieWhere, options: MovieOptions): [Movie!]!
+  moviesAggregate(where: MovieWhere): MovieAggregateSelection!
+  moviesConnection(first: Int, after: String, where: MovieWhere, sort: [MovieSort]): MoviesConnection!
+}
+
+type Mutation {
+  updateMovies(where: MovieWhere, update: MovieUpdateInput): UpdateMoviesMutationResponse!
+}
+
+type Subscription {
+  movieCreated(where: MovieSubscriptionWhere): MovieCreatedEvent!
+  movieUpdated(where: MovieSubscriptionWhere): MovieUpdatedEvent!
+  movieDeleted(where: MovieSubscriptionWhere): MovieDeletedEvent!
+}
+```
+
+In other words, the `createMovies` and `deleteMovies` mutations have not been generated, the query type was not impacted because it does not do any create/ delete operations and the subscriptions were not impacted because of the current behavior.\
+\
+\
+The proposed solution would would make use of a new operation in the exclude list:
+
+```graphql
+type Movie @exclude(operations: [SUBSCRIPTIONS, CREATE, DELETE]) {
+  title: String!
+  released: Int!
+}
+```
+
+and have as outcome the exclusion of the subscription fields altogether:
+
+```graphql
+type Query {
+  movies(where: MovieWhere, options: MovieOptions): [Movie!]!
+  moviesAggregate(where: MovieWhere): MovieAggregateSelection!
+  moviesConnection(first: Int, after: String, where: MovieWhere, sort: [MovieSort]): MoviesConnection!
+}
+
+type Mutation {
+  updateMovies(where: MovieWhere, update: MovieUpdateInput): UpdateMoviesMutationResponse!
+}
+
+type Subscription {
+  <!-- no fields generated -->
+}
+```
+
+##### Out of Scope
+
+1. At this point we are **not** yet considering fine-grain control on excluding specific subscription operations. Either all subscription operations on a specified type are excluded, or none. This means nothing like this will be possible for now:
+
+```graphql
+type Movie @exclude(operations: [SUBSCRIPTIONS_CREATE]) {
+  ...
+}
+<!-- generating.. -->
+type Subscription {
+  <!-- no create operation -->
+  movieUpdated(where: MovieSubscriptionWhere): MovieUpdatedEvent!
+  movieDeleted(where: MovieSubscriptionWhere): MovieDeletedEvent!
+}
```

Comment:
**Not** important, but technically comments in graphql are done with #, which may help if we copy-paste these examples at some point

```suggestion
type Movie @exclude(operations: [SUBSCRIPTIONS_CREATE]) {
  ...
}
# generating..
type Subscription {
  # no create operation
  movieUpdated(where: MovieSubscriptionWhere): MovieUpdatedEvent!
  movieDeleted(where: MovieSubscriptionWhere): MovieDeletedEvent!
}
```